app/src/main/java/com/example/amcc/view/MainFragment.java

```diff
@@ -72,9 +80,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 
         navController = Navigation.findNavController(view);
         view.findViewById(R.id.btnResultID).setOnClickListener(v -> {
-            CarDetails car = new CarDetails(citySelected, Integer.parseInt(engineSizeField.getText().toString()), Integer.parseInt(emissionEdtField.getText().toString()), FuelType.e5, regDate, Integer.parseInt(avgConField.getText().toString()), Integer.parseInt(milePerYField.getText().toString()));
-            viewModel.setApiData(car);
-            navController.navigate(R.id.resultFragment);
+
+            validateInput();
+            if (validation.validate()) {
+                CarDetails car = new CarDetails(citySelected,
+                        Integer.parseInt(engineSizeField.getText().toString()),
+                        Integer.parseInt(emissionEdtField.getText().toString()),
+                        FuelType.e5, regDate,
```

Comment:
@Jaber-aldroubi I should say, FuelType is useless! String variable should be enough, bec. we have fixed RadioButton  input.